### PR TITLE
Week 2 and Week 3 Homework and Questions

### PR DESCRIPTION
**Pull request by Mattsface**
How's the homework look? Is that what you are looking for?

It's the last commit...

### CODE REVIEW

File: week1/exercises/rspec_spec.rb [hunk 1]
```diff
@@ -73,6 +73,25 @@
   end
 
   context "Examples for in-class test exploration" do
+<<<<<<< HEAD
```
Comment (bethanyr) [inline week1/exercises/rspec_spec.rb]: when you see the <<<<<< in your files - it means that you had a merge conflict with git. git then adds in both versions of the file so you can choose which version you should keep (in most cases git can figure out how to merge 2 versions of files, but in your case it couldn't). You should delete out the version that you do not want to keep. This section is what is on master.

File: week1/exercises/rspec_spec.rb [hunk 2]
```diff
@@ -89,6 +108,7 @@
 
     it "should check basic spelling" do
       "Field".should include('ie')
+>>>>>>> 4981dd20c06248919b350fa8600269a41e169e64
```
Comment (bethanyr) [inline week1/exercises/rspec_spec.rb]: this is the other version of the code that git could not merge in your file.

File: week1/homework/questions.txt [hunk 3]
```diff
@@ -4,12 +4,43 @@ p.86-90 Strings (Strings section in Chapter 6 Standard Types)
...
+
+str.downcase
+	returns a copy of the string where every uppercase character in the string is a lowercase 
+
+str.hash
+	returns a hash of the string
+
 6. What are two ways of defining a String literal? Bonus: What is the difference between them?
+
+Good lord, I've never seen so many ways to define a string literal in a language before. What is the best practice here?
```
Comment (bethanyr) [inline week1/homework/questions.txt:34]: usually developers use "" and '' to define strings.

File: week2/homework/questions.txt [hunk 4]
```diff
@@ -4,10 +4,22 @@ Sharing Functionality: Inheritance, Modules, and Mixins
...
 
+	Arrays are used when speed is important and elements need to be organized in an ordered fashion. Hashes are used when data needs to be accessed by a key, not the index. 
+
 3. What is a module? Enumerable is a built in Ruby module, what is it?
 
+	A module is a library of reusable code that can be imported into other code. Modules can contain methods, classes, and constants. Emumerable is module that allows your classes to support many of Rubys popular methods like ".include?"
+
 4. Can you inherit more than one thing in Ruby? How could you get around this problem?
 
+	Yes you can, well, I'm just throwing this out there, I'm probably wrong :), but from my understanding inheritance looks like it works in a hierarchical fashion. So if you want to inherit from more than one class, you'll have to structure your classes in a manner that allow the inheritance to flow up the chain. I'm having trouble articulating this, so I'll try and draw it.
```
Comment (bethanyr) [inline week2/homework/questions.txt:19]: Ruby only supports single inheritance - what you illustrate is correct and how single inheritance works in Ruby.

File: week2/homework/simon_says.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+module SimonSays
+
+	# return the string
+	def echo(string)
```
Comment (bethanyr) [inline week2/homework/simon_says.rb:4]: in ruby - you don't need #{} to reference variables.

File: week3/homework/calculator.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	# method to check if args passed to method is a multidimensional array
+	def is_dim(var)
+		var.each do |x| 
+			true if x.is_a?(Array)
+		end 
+		false
+	end
+
+	def sum(*args)
+		args.flatten! if is_dim(args) # flatten array if its multidimensional
```
Comment (bethanyr) [inline week3/homework/calculator.rb:12]: when I run the calculator tests against your code, I get 4 failing tests for sum and multiply. With flatten, you don't have to check if it is a multi-dimensional array. try taking out the check for is_dim in your code and see if they pass for you.

**Comment by bethanyr** [comment]
I reviewed week 2 and week 3 homework - looks good.

File: week3/homework/calculator.rb (hunk 6 above)
Comment (Mattsface) [inline week3/homework/calculator.rb:12]: Odd... I could have swore it was working last night. I removed that code now. 

Is it okay to try and flatten an array even if I don't know its multi-dimensional?

**Pull request closed without merging**